frostsnap_core/src/coordinator.rs

```diff
@@ -495,53 +496,73 @@ impl FrostCoordinator {
                 keygen_id,
                 ack_session_hash,
             }) => {
-                let mut outgoing = vec![];
-                match self.pending_keygens.get_mut(&keygen_id) {
-                    Some(KeyGenState::WaitingForAcks {
-                        agg_input,
-                        device_to_share_index,
-                        acks,
-                        ..
-                    }) => {
-                        let session_hash = SessionHash::from_agg_input(agg_input);
+                let wrong_state_err = Error::coordinator_invalid_message(
```

Comment:
:pick: Better return the same error from different branches so it's easier to debug